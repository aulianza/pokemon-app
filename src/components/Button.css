.btn {
    display: -webkit-inline-box;
    display: inline-flex;
    font-family: "Rubik", sans-serif;
    font-size: 1.1em;
    font-weight: 500;
    padding: 10px 25px;
    background-color: #607d8b;
    border: 2px solid #607d8b;
    border-width: 1px;
    border-style: solid;
    border-color: rgba(0, 0, 0, 0);
    border-radius: 4px;
    cursor: pointer;
    -webkit-box-pack: center;
    justify-content: center;
    text-align: center;
    letter-spacing: inherit;
    white-space: nowrap;
    color: #fff;
    transition: 0.3s;
    margin: 0 5px;
}

.btn:hover {
    background-color: #fff;
    transition: 0.3s;
}

.btn-block {
    display: block;
    width: 100%;
    padding-right: 0;
    padding-left: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
}

.btn-danger {
    background-color: #e63757;
    border: 2px solid #e63757;
}

.btn-danger:hover {
    color: #e63757;
    border: 2px solid #e63757;
}

.btn-primary {
    background-color: #3761ab;
    border: 2px solid #3761ab;
}

.btn-primary:hover {
    color: #3761ab;
    border: 2px solid #3761ab;
}

.btn-success {
    background-color: #4caf50;
    border: 2px solid #4caf50;
}

.btn-success:hover {
    color: #4caf50;
    border: 2px solid #4caf50;
}
