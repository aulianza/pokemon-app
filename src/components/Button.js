import React from "react";

import "./Button.css";

const Button = (props) => {
    const classes = ["btn"];

    if (props.buttonColor) classes.push(props.buttonColor);
    if (props.isBlock) classes.push("btn-block");

    const clickHandler = () => {
        props.onButtonClicked();
    }

    return <button className={classes.join(" ")} onClick={clickHandler}>{props.title}</button>;
};

export default Button;
