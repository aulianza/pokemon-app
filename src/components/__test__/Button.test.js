import Button from "../Button";
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("button-component, snapshot", () => {
    act(() => {
        render(<Button />, container)
    })
    expect(container).toMatchSnapshot();
})

it("should render button correctly", () => {
    act(() => {
        render(<Button title="Catch Now!" />, container)
    })
    expect(container).toBeTruthy();
});
