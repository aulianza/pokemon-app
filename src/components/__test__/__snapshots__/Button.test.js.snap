// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button-component, snapshot 1`] = `
<div>
  <button
    class="btn"
  />
</div>
`;
